### PR TITLE
angle_function

### PR DESCRIPTION
**Pull request by satvik-dixit**
This Pull request modifies `angle_function` for computing the azimuth and elevation between a set of points and a single point. This will be useful for incorporating directivities, i.e. for PR #175.

---

Thanks for sending a pull request (PR), we really appreciate that! Before hitting the submit button, we'd be really glad if you could make sure all the following points have been cleared.

Please also refer to the doc on [contributing](https://pyroomacoustics.readthedocs.io/en/pypi-release/contributing.html) for more details. Even if you can't check all the boxes below, do not hesitate to go ahead with the PR and ask for help there.

- [x] Are there docstrings ? Do they follow the [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard) style ?
- [x] Have you run the tests by doing `nosetests` or `py.test` at the root of the repo ?
- [ ] Have you checked that the doc builds properly and that any new file has been added to the repo ? How to do that is covered in the [documentation](https://pyroomacoustics.readthedocs.io/en/pypi-release/contributing.html#documentation).
- [x] Is there a unit test for the proposed code modification ? If the PR addresses an issue, the test should make sure the issue is fixed.
- [ ] Last but not least, did you document the proposed change in the CHANGELOG file ? It should go under "Unreleased".

Happy PR :smiley:

### CODE REVIEW

File: pyroomacoustics/tests/test_angle_function.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        self.assertTrue(angle_function(c2,d1).all()==np.array([[0,0],[-pi/2,0],[pi/4,0]]).all())
+        self.assertTrue(angle_function(c2,d2).all()==np.array([[-pi/4,0],[-pi/2,0],[-pi/2,0]]).all())
+        self.assertTrue(angle_function(c2,d3).all()==np.array([[pi/2,0],[pi,0],[3*pi/4,0]]).all())
+        self.assertTrue(angle_function(c3,d1).all()==np.array([[pi/4,0]]).all())
+        self.assertTrue(angle_function(c3,d2).all()==np.array([[0,0]]).all())
+        self.assertTrue(angle_function(c3,d3).all()==np.array([[pi/2,0]]).all())
+
+
+if __name__=='__main__':
+    unittest.main()
```
Comment (ebezzam) [inline pyroomacoustics/tests/test_angle_function.py:50]: here you can call a separate function within which you call `angle_function` like [this](https://github.com/LCAV/pyroomacoustics/blob/fe80a7332ce2ca1e592a62a078bec621750a0ad9/pyroomacoustics/tests/test_autocorr.py#L52)

**Review by ebezzam — CHANGES_REQUESTED** [review]

File: pyroomacoustics/tests/test_angle_function.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        for c in c_range:
+            for d in d_range:
+
+                error_azimuth=(angle_function(c,d) - c_d_range[c_d_index])[:,0]
+                error_elevation=(angle_function(c,d) - c_d_range[c_d_index])[:,1]
+
+                print("for points :")
+                print(c,d)
+                print("error in azimuth calculation: {}".format(np.average(error_azimuth)))
+                print("error in elevation calculation: {}".format(np.average(error_elevation)))
```
Comment (ebezzam) [inline pyroomacoustics/tests/test_angle_function.py:113]: Rename all elevation to colatitude

Comment (satvik-dixit) [reply pyroomacoustics/tests/test_angle_function.py]: renamed elevation to colatitude in angle_function and test_angle_function

File: pyroomacoustics/utilities.py [hunk 3]
```diff
@@ -743,5 +743,49 @@ def goertzel(x, k):
...
+    # colatitude calculation for 2-D coordinates
+    elif s1.shape[1] == 2 and v2.shape[0] == 2:
+                                                                 
+        num_points = s1.shape[0]
+        colatitude = np.ones(num_points, dtype=int)*np.pi/2
+
+    # azimuth calculation (same for 2-D and 3-D)
+    azimuth = np.arctan2((y_vals-y2), (x_vals-x2))
+
+    return np.vstack((azimuth, colatitude)).T
```
Comment (ebezzam) [inline pyroomacoustics/utilities.py:791]: how about [hstack](https://numpy.org/doc/stable/reference/generated/numpy.hstack.html) to avoid the transpose?

**Comment by ebezzam** [comment]
Covered by #231

**Pull request closed without merging**